qubesappmenus/__init__.py

```diff
@@ -249,6 +251,11 @@ def appmenus_create(self, vm, force=False, refresh_cache=True):
             anything_changed = True
             directory_changed = True
         appmenus = list(self.get_available_filenames(vm))
+        def make_desktop_name(vm, appmenu_basename):
+            return '.'.join((qubes_vm_desktop, vm.name, appmenu_basename))
+        def make_settings_name(vm):
+            return '.'.join((qubes_vm_desktop_settings, vm.name, 'desktop'))
```

Comment:
Better make those class functions instead of local functions. This will require making `qubes_vm_desktop*` variables class variables too, which improves readability too.
In fact, you do have this function duplicated in tests already.